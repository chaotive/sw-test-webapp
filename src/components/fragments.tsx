import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import React from 'react'
import Spinner from 'react-bootstrap/Spinner'

export const loadingFragment = <Row><Col className="text-center m-4">
  <Spinner animation="border" role="status">
    <span className="sr-only">Loading...</span>
  </Spinner>
</Col></Row>

export const titleFragment = (text: string) => <Row>
  <Col as="h1">{text}</Col>
</Row>
