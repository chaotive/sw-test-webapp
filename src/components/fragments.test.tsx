import ReactDOM from 'react-dom'
import {loadingFragment, titleFragment} from './fragments'
import renderer from 'react-test-renderer'

describe('loadingFragment', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(loadingFragment, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders correctly', () => {
    const tree = renderer
      .create(loadingFragment)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('titleFragment', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(titleFragment('someTitle'), div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders correctly edge case', () => {
    const tree = renderer.create(titleFragment('')).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly regular case', () => {
    const tree = renderer.create(titleFragment('someTitle')).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
