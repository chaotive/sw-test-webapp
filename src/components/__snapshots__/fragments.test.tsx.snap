// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadingFragment renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="text-center m-4 col"
  >
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`titleFragment renders correctly edge case 1`] = `
<div
  className="row"
>
  <h1
    className="col"
  >
    
  </h1>
</div>
`;

exports[`titleFragment renders correctly regular case 1`] = `
<div
  className="row"
>
  <h1
    className="col"
  >
    someTitle
  </h1>
</div>
`;
